### DISCUSSION TITLE
Scrolling text in editor up or down one line

### DISCUSSION THREAD

**Discussion by JimPettis** [atom/atom#22417] [Q&A]
Feature common in Windows-based code & HTML editors: scroll the entire (displayed) editor screen up or down by one line. Normally the key-combination used is CTRL-UP or CTRL-DOWN.

This is an extremely useful feature.

Atom does something really odd (in my Windows- and DOS- based experience) with CTRL-UP and CTRL-DOWN: it moves the current line above the preceding line and below the following line (respectively).

SO:
Is there any keycode in Atom to scroll the editor text up or down by one line?
If not, is there a such a function available, and can it be "bound" to a keystroke combination?

**Comment by sakkamade** [comment]
You should try "scroll-with-keeping-cursor-pos" package, and set its commands in `keymap` file to desired by you <kbd>Ctrl</kbd>+<kbd>Up</kbd> and <kbd>Ctrl</kbd>+<kbd>Down</kbd>, like so:
```
'atom-text-editor':
  'ctrl-down': 'scroll-with-keeping-cursor-pos:scroll-down'
  'ctrl-up': 'scroll-with-keeping-cursor-pos:scroll-up'
```
